### PR TITLE
feat: Support gRPC Compression

### PR DESCRIPTION
**Pull request by yifatgortler**
Enable gRPC compression when a compressor name is provided to JsonStreamWriter/StreamWriter.
Only `gzip` compression is supported at the moment.

Please note that while it will reduce the network usage, it will increase CPU usage. Please verify the trade off per use case.

### CODE REVIEW

File: google-cloud-bigquerystorage/src/main/java/com/google/cloud/bigquery/storage/v1/JsonStreamWriter.java [hunk 1]
```diff
@@ -336,6 +337,22 @@ public Builder setLocation(String location) {
...
+     * Sets the compression to use for the calls. The compressor must be of type gzip.
+     *
+     * @param compressorName
+     * @return Builder
+     */
+    public Builder setCompressorName(String compressorName) {
+      Preconditions.checkNotNull(compressorName);
+      Preconditions.checkArgument(
+          compressorName.equals("gzip"),
+          "Compression of type \"%s\" isn't supported, only \"gzip\" compression is supported.",
```
Comment (yirutang) [inline google-cloud-bigquerystorage/src/main/java/com/google/cloud/bigquery/storage/v1/JsonStreamWriter.java:350]: Only check it at the bottom layer (ConnectionWorker) is enough?

File: google-cloud-bigquerystorage/src/main/java/com/google/cloud/bigquery/storage/v1/SchemaAwareStreamWriter.java [hunk 2]
```diff
@@ -609,6 +616,22 @@ public Builder<T> setLocation(String location) {
...
 
+    /**
+     * Sets the compression to use for the calls. The compressor must be of type gzip.
+     *
+     * @param compressorName
+     * @return Builder
+     */
+    public Builder<T> setCompressorName(String compressorName) {
+      Preconditions.checkNotNull(compressorName);
+      Preconditions.checkArgument(
```
Comment (yirutang) [inline google-cloud-bigquerystorage/src/main/java/com/google/cloud/bigquery/storage/v1/SchemaAwareStreamWriter.java:627]: Same as above.

**Review by yirutang — COMMENTED** [review]
Thanks for getting this fixed!

File: google-cloud-bigquerystorage/src/main/java/com/google/cloud/bigquery/storage/v1/JsonStreamWriter.java (hunk 1 above)
Comment (yifatgortler) [reply google-cloud-bigquerystorage/src/main/java/com/google/cloud/bigquery/storage/v1/JsonStreamWriter.java]: Removed from JsonStreamWriter and SchemaAwareStreamWriter, and left it in StreamWriter. Verified that it's what we usually do.

**Review by yirutang — APPROVED** [review]

**Pull request merged**